src/compiler/checker.ts

```diff
@@ -351,10 +336,14 @@ namespace ts {
             isUndefinedSymbol: symbol => symbol === undefinedSymbol,
             isArgumentsSymbol: symbol => symbol === argumentsSymbol,
             isUnknownSymbol: symbol => symbol === unknownSymbol,
-            getMergedSymbol,
+            getMergedSymbol: symbol => {
+                verifyTypeCheckerOfSymbol(symbol);

```

Comment:
How about having `verifyCheckerOfXXX` return the value it is passed such that you can write `return getMergedSymbol(verifyCheckerOfSymbol(symbol))` and keep it to just one expression.